scio-test/src/test/scala/com/spotify/scio/coders/CoderTest.scala

```diff
@@ -43,161 +58,100 @@ import java.time.{Instant, LocalDate}
-final case class TB(aDouble: Double) extends Top
+  val userId: UserId = UserId(Seq[Byte](1, 2, 3, 4))
+  val user: User = User(userId, "johndoe", "johndoe@spotify.com")
 
-// case classes
-final case class DummyCC(s: String)
-final case class ParameterizedDummy[A](value: A)
-final case class MultiParameterizedDummy[A, B](valuea: A, valueb: B)
+  def materialize[T](coder: Coder[T]): BCoder[T] =
+    CoderMaterializer.beam(PipelineOptionsFactory.create(), coder)
 
-// objects
-object TestObject
-object TestObject1 {
-  val somestring = "something"
-  val somelong = 42L
-}
+  it should "support primitives" in {
+    false coderShould roundtripToBytes(Array(0)) and
+      beOfType[Beam[_]] and
+      materializeTo[BooleanCoder]
 
-// explicit coder
-case class CaseClassWithExplicitCoder(i: Int, s: String)
-object CaseClassWithExplicitCoder {
-  import org.apache.beam.sdk.coders.{AtomicCoder, StringUtf8Coder, VarIntCoder}
-  import java.io.{InputStream, OutputStream}
-  implicit val caseClassWithExplicitCoderCoder: Coder[CaseClassWithExplicitCoder] =
-    Coder.beam(new AtomicCoder[CaseClassWithExplicitCoder] {
-      val sc = StringUtf8Coder.of()
-      val ic = VarIntCoder.of()
-      def encode(value: CaseClassWithExplicitCoder, os: OutputStream): Unit = {
-        ic.encode(value.i, os)
-        sc.encode(value.s, os)
-      }
-      def decode(is: InputStream): CaseClassWithExplicitCoder = {
-        val i = ic.decode(is)
-        val s = sc.decode(is)
-        CaseClassWithExplicitCoder(i, s)
-      }
-    })
-}
+    1 coderShould roundtripToBytes(Array(1)) and
+      beOfType[Beam[_]] and
+      materializeTo[VarIntCoder]
 
-// nested
-case class NestedB(x: Int)
-case class NestedA(nb: NestedB)
+    1L coderShould roundtripToBytes(Array(0, 0, 0, 0, 0, 0, 0, 1)) and
+      beOfType[Beam[_]] and
+      materializeTo[BigEndianLongCoder]
 
-// recursive
-case class SampleField(name: String, fieldType: SampleFieldType)
-sealed trait SampleFieldType
-case object IntegerType extends SampleFieldType
-case object StringType extends SampleFieldType
-case class RecordType(fields: List[SampleField]) extends SampleFieldType
-case class Recursive(a: Int, rec: Option[Recursive] = None)
+    'a' coderShould roundtripToBytes(Array(97)) and
+      beOfType[Transform[_, _]] and
+      materializeToTransformOf[ByteCoder]
 
-// private
-class PrivateClass private (val value: Long) extends AnyVal
-object PrivateClass {
-  def apply(l: Long): PrivateClass = new PrivateClass(l)
-}
-case class UsesPrivateClass(privateClass: PrivateClass)
+    "yolo" coderShould roundtripToBytes(Array(121, 111, 108, 111)) and
+      beOfType[Beam[_]] and
+      materializeTo[StringUtf8Coder]
 
-// proto
-case class ClassWithProtoEnum(s: String, `enum`: OuterClassForProto.EnumExample)
+    4.5 coderShould roundtripToBytes(Array(64, 18, 0, 0, 0, 0, 0, 0)) and
+      beOfType[Beam[_]] and
+      materializeTo("SDoubleCoder$") and
```

Comment:
not a fan of the string param here. We can probably work out smth

Reply:
@RustedBones so now I have changed `private` to `private[coders]` for multiple classes to avoid this

Reply:
Yes, package visibility for coders sounds appropriate